### PR TITLE
Avoid duplicate avatar images

### LINKED ISSUE #16721: Reader: Post has duplicate gravatar
I don't know now if it's a feature or bug, but the `.author-compact-profile` has two gravatar images.

ss_2017-07-30-13-24-38

Same in WordPress tablet app

screenshot_2017-07-30-12-00-43

### PR DESCRIPTION
**Pull request by mikaelz**
Should fix #16721

**Test**
1. Visit http://calypso.localhost:3000/ and check avatar appears?
2. Random click on author name. Should be at `http://calypso.localhost:3000/read/feeds/:feed-id` (e.g. http://calypso.localhost:3000/read/feeds/52425000) and check avatars do appear?
3. Random click post title (e.g. http://calypso.localhost:3000/read/feeds/52425000/posts/1442165915). Check avatar does appear in sidebar?

### CODE REVIEW

**Comment by alisterscott** [comment]
Thanks for your contribution @mikaelz 

There are some circumstances where the images are different, for example sites that have a lot of different authors.

For example, this post https://wordpress.com/read/feeds/24643651/posts/1536780728

Before this PR:

screen shot 2017-07-31 at 1 10 03 pm

With this PR:

screen shot 2017-07-31 at 1 10 09 pm

I have asked for some design input from our designers to see what the desired visual appearance is in this circumstance.

**Comment by bluefuton** [comment]
Thanks for the contribution @mikaelz!

As @alisterscott says, there are many cases where the two icons are different, and we want to continue to display both icons in those cases.

The challenge is finding out when the two are actually different. They might have different filenames, but visually appear the same. The comparison could either happen on the API side or with a library like http://huddle.github.io/Resemble.js/, but it feels pretty heavy in both cases.

**Comment by samouri** [comment]
i looked into resemblejs many moons ago: https://github.com/Automattic/wp-calypso/issues/8659#issuecomment-258065709

tldr: it seemed like a dead end for solving this issue

**Comment by fraying** [comment]
I agree that seeing the same image twice is a problem, but it's an edge case. For it to happen, the user had to add the same image as their gravatar and their site icon. Usually they're different.

**Comment by mikaelz** [comment]
I just uploaded my photo at https://wordpress.com/me (no option for removal) and have Gravatar set up for email. From my view it's a bit distracting the two images (site icon from https://wordpress.com/me and Gravatar for email).

Other examples
https://wordpress.com/read/feeds/52425000/posts/1544634615
https://wordpress.com/read/blogs/22158974/posts/8687
https://wordpress.com/read/blogs/63023584/posts/2272
https://wordpress.com/read/blogs/51877888/posts/367
https://wordpress.com/read/feeds/23259639/posts/1544168568 (Jeremy knew how to set up :)
for me it looks kinda buggy the double images.
Checked Matt's https://wordpress.com/read/feeds/188407/posts/1486029391 and it's OK. Don't know how he achieved it :D

What bothers me is the psychological perception from it :) I would be interested in some user tests how they perceive it, feature or bug?

I'm more towards sticking to one image. I think there are 2 types of people who care (designer likes) and who care not (no fixing needed). With 1 image the designer like wouldn't perceive it as bug.

**Comment by fraying** [comment]
Thanks for the attention, @mikaelz, but this is as designed, so closing this for now.
